pkg/generate/generate.go

```diff
@@ -129,10 +129,10 @@ func (c *Controller) applyGenerate(resource unstructured.Unstructured, gr kyvern
 	}
 	engineResponse.PolicyResponse.Rules = []response.RuleResponse{}
 
-	for _, v := range policy.Spec.Rules {
+	for _, v := range policyObj.Spec.Rules {
 		for _, r := range rules {
-			if policy.Name == engineResponse.PolicyResponse.Policy && r.Name == v.Name {
-				if len(v.MatchResources.ResourceDescription.Kinds) > 0 && (len(v.MatchResources.ResourceDescription.Annotations) == 0 || len(v.MatchResources.ResourceDescription.Selector.MatchLabels) == 0) {
+			if policyObj.Name == engineResponse.PolicyResponse.Policy && r.Name == v.Name && len(v.MatchResources.Kinds) > 0 {
+				if len(v.MatchResources.Annotations) > 0 || v.MatchResources.Selector == nil {
```

Comment:
Shouldn't it be `v.MatchResources.Selector != nil`?

Reply:
no we need nil condition for backword compatibility

Reply:
In general, when there's a matched annotation or label selector,  we would want to apply the policy. 

But here, `len(v.MatchResources.Annotations) > 0` means annotation selector exist; `v.MatchResources.Selector == nil`  means selector does not exist - isn't these 2 conditions contrast with each other?

Reply:
yes it should be `len(v.MatchResources.Annotations) == 0`